### PR TITLE
[3.10] gh-97032: avoid test_squeezer crash on macOS buildbots

### PR DESCRIPTION
**Pull request by ned-deily**
!buildbot macOS

* Issue: gh-97032

### CODE REVIEW

**Comment by ned-deily** [comment]
!buildbot x86-64 macOS PR

**Comment by ned-deily** [comment]
!buildbot ARM64 MacOS

**Comment by ned-deily** [comment]
@encukou @ambv

**Comment by encukou** [comment]
!buildbot ARM64 MacOS

**Comment by encukou** [comment]
PR buildbots for old branches aren't particularly stable, so it's better to look at the issues and ignore unrelated ones. 
There's no tkinter-related crash here, so I'd count them as successful.

@pablogsal, IMO this is the best way to fix the 3.10 & 3.9 buildbot failures. There's a more robust fix in later branches. 
The other approaches considered are:
- change library detection logic, which isn't ideal in source-only security releases
- configuring the buildbot to use Homebrew tkinter -- too much work for a library that isn't really tested anyway

File: Lib/idlelib/idle_test/test_squeezer.py [hunk 1]
```diff
@@ -170,6 +170,7 @@ def test_write_not_stdout(self):
 
     def test_write_stdout(self):
         """Test Squeezer's overriding of the EditorWindow's write() method."""
+        requires('gui')
```
Comment (terryjreedy) [inline Lib/idlelib/idle_test/test_squeezer.py:173]: It makes no sense to me that this should have any effect as this test should not require a gui and does not on all other CI and buildbots that do not have such.  Indeed, if Apple tkinter is being used, I expect `requires('gui')` should be True, and this should also have no effect.  This could be conditioned on `if macosx.isAquaTk():`, which is already used elsewhere in this file.

**Review by terryjreedy — COMMENTED** [review]
Running IDLE on Mac with Cocoa and Apple tkinter is not supported.  When I user does so, IDLE warns about stability and points to the web page about installing better tkinter.

The crash makes no sense to me; do whatever whoever thinks best.

File: Lib/idlelib/idle_test/test_squeezer.py (hunk 1 above)
Comment (ned-deily) [reply Lib/idlelib/idle_test/test_squeezer.py:173]: Yeah, I was confused by this, too. But it was clear from the buildbot log that all the other squeezer tests were being properly skipped due to failing their `requires('gui')` check as expected. The one thing unique to `test_write_stdout` is that it has two calls to `make_squeezer_instance()` and it is the second call that doesn't properly fail, i.e. it proceeds to actually create a real Tk() root and that's what causes the segfault under these circumstances. As best I can tell, that is still happening on `main` and such but, with an up-to-date Tk 8.6.13(), that call doesn't cause a crash anymore. An edge case, in any case, and it really only comes up because of this one buildbot configuration.

**Comment by encukou** [comment]
The test failure is SIGABRT that, AFAIK, comes from inside tkinter.
```
macOS 12 (1207) or later required, have instead 12 (1206) !
Fatal Python error: Aborted
```

That is, the tkinter that comes with the system now crashes the process at startup. AFAIK, this started happening after a system upgrade.
The right thing Python could do about this is not select this tkinter in `./configure`. Later branches do this; for 3.9-3.10, changing the logic would IMO be too invasive.
Hence this PR: avoid importing tkinter at all if we're not testing GUI.

**Comment by pablogsal** [comment]
While I understand the fix and the rationale, I have to say that I still find it a bit "hacky" (for the lack of a better term). Given that this is to fix CI in old branches I am fine merging it as it has no user impact, but ideally we should fix this in the build system in `main` if we detect that we are using an incompatible `tkinter `.

I any case, I am merging this as I think the proper fix will be more aggressive and we need to have the CI green to avoid hiding other problems. Thanks @encukou and @ned-deily for the PR and the explanation.

**Review by pablogsal — APPROVED** [review]

**Comment by encukou** [comment]
Thanks!
I agree it's hacky, but I don't see a way toward a better fix than what's in `main` now -- using Tk that the user has control over, and making them responsible for providing something that doesn't crash.
I don't think we want e.g. to try creating GUI windows in `configure`/`make` to see if Tk crashes.
